frontend/src/components/object/ObjectFileDetails.vue

```diff
@@ -2,25 +2,42 @@
 import { onMounted, ref } from 'vue';
 import { useRoute } from 'vue-router';
 import { storeToRefs } from 'pinia';
+import Button from 'primevue/button';
+import Dialog from 'primevue/dialog';
 import { useToast } from 'primevue/usetoast';
 
-import ObjectAccess from '@/components/object/ObjectAccess.vue';
-import ObjectMetadata from '@/components/object/ObjectMetadata.vue';
-import ObjectProperties from '@/components/object/ObjectProperties.vue';
-import ObjectTag from '@/components/object/ObjectTag.vue';
-import { useObjectStore } from '@/store';
+import {
+  DeleteObjectButton,
+  DownloadObjectButton,
+  ObjectAccess,
+  ObjectMetadata,
+  ObjectPermission,
+  ObjectProperties,
+  ObjectTag
+} from '@/components/object';
+
+import { useObjectStore, useUserStore } from '@/store';
+import { Permissions } from '@/utils/constants';
 
 import type { Ref } from 'vue';
 import type { COMSObject } from '@/interfaces';
+import { ButtonMode } from '@/interfaces/common/enums';
 
+const objectStore = useObjectStore();
 const toast = useToast();
 const route = useRoute();
 
+const { objectList } = storeToRefs(objectStore);
+const { currentUser } = storeToRefs(useUserStore());
+
 const objectInfo: Ref<COMSObject> = ref({} as COMSObject);
 const isInfoLoaded: Ref<Boolean> = ref(false);
-const { objectList } = storeToRefs(useObjectStore());
 
-const getObjectInfo = async (objId: string) => {
+const permissionsVisible = ref(false);
+const permissionsObjectId = ref('');
+const permissionsObjectName = ref('');
```

Comment:
TS Style: We'll likely want to be explicit about the Ref template typings here (Ref<boolean>, Ref<string>, etc)